sympy/solvers/solvers.py

```diff
@@ -1968,7 +1998,7 @@ def key(sym):
     # so the docs should be updated to reflect that or else
     # the following should be `bool(failed) or not linear`
     default_simplify = bool(failed)
-    if flags.get('simplify', default_simplify):
+    if flags['simplify'] is True or flags['simplify'] is None and default_simplify:
```

Comment:
up to here, keywords did not have to be sanitized because `in` was used; but `is` is strict:
```python
>>> 1 in (True,)
True
>>> 1 is True
False
```

Reply:
I'll change it to use `==`. Generally I don't see the value in allowing `1` instead of `True` but the intention of this PR is not to change any observable behaviour.

Reply:
For those of us who use 0,1 interchangeably for False,True it will make a difference. Thanks.

I couldn't find the issue where an options handling routine (like used by Polys) was discussed. But it seems like such a thing should be added at some point, especially when there are keywords for different routines being handled.

Reply:
I think it would be better to just reduce the number of options. Many of these things would be better just as separate functions rather than options to solve. The control flow of solve would be a lot easier to understand and probably less fragile if it didn't depend on manipulating and passing around so many options.